meilisearch/client.py

```diff
@@ -161,6 +161,25 @@ def health(self):
         """
         return self.http.get(self.config.paths.health)
 
+    def is_healthy(self):
+        """Get health of the MeiliSearch server.
+
+        `200` HTTP status response when MeiliSearch is healthy.
+
+        Return
+        ------
+        health: True | False
+        """
+        try:
+            self.http.get(self.config.paths.health)
+        except MeiliSearchApiError as err:
+            raise err
+        except MeiliSearchCommunicationError as err:
+            raise err
+        except MeiliSearchTimeoutError as err:
+            raise err
```

Comment:
For me, you need to catch any kind of error, not this 3 ones ^^
Plus, the is_healthy method should not raise an error, but return false 🙂

Reply:
For me, and to avoid maintenance, you should:
```
except Exception as e:
```

or maybe only `MeiliSearchApiError` and `MeiliSearchCommunicationError` but `MeiliSearchTimeoutError` does not make any sense for health :)

Reply:
We can't afford to add every new custom errors here 🙂
is_health returns only true or false. If the users want the error, they use `health()`

Reply:
It was my first writing! but I had troubles with the linter I thought it came from there. My bad I prefer this way too

Reply:
I was right I already tried this solution, but I can resolve the linter problem ...
```
try:
    self.health()
except Exception:
    return False
```
It's says:
```
************* Module meilisearch.client
meilisearch/client.py:175:15: W0703: Catching too general exception Exception (broad-except)
```
Did you know how to deal with it I not sure it's a good idea to change the linter option ...

Reply:
Two solutions:
- either we disable pylint for this rule and this line -> https://stackoverflow.com/questions/28829236/is-it-possible-to-ignore-one-single-specific-line-with-pylint
- either we catch only the `MeiliSearchError`, since all our errors inherits for the `MeiliSearchError` class, but pretty sure it will not work

Reply:
My guess is that it is simple to change the linter rules than the upstream functions which would induce a code revision. Don’t you ?

Reply:
`requests.exceptions.RequestException` is the base exception in requests so catching this should catch all the errors from requests and avoid the pylint error with catching the generic `Exception`. The reason pylint complains about catching the generic `Exception` is it can mask unrelated errors making debugging harder.

@curquiza obviously it's you call on the best way to handle this. @alallema you can tell pylint to ignore the generic exception rule on this line with:

```py
try:
    self.health()
except Exception:  # pylint: disable=broad-except
    return False
```

Reply:
@sanders41 Thanks a lot! Its works perfectly. Anyway, I change the `Exception` by general `MeiliSearchError` and it seems to work well too.

Reply:
I think I would rather the `requests.exceptions.RequestException` to be sure we can catch all the exceptions from request :)
I'm afraid MeiliSearchError is not exhaustive, WDYT @alallema?

Reply:
(And thanks @sanders41 for your help!!)

Reply:
@curquiza I'm not sure either, but there are already catching in upstream functions:
```py
try:
    ...
except requests.exceptions.Timeout as err:
     raise MeiliSearchTimeoutError(err) from err
except requests.exceptions.ConnectionError as err:
     raise MeiliSearchCommunicationError(err) from err
```
```py
try:
     request.raise_for_status()
     return HttpRequests.__to_json(request)
except requests.exceptions.HTTPError as err:
    raise MeiliSearchApiError(err, request) from err
```
So we can do a code revision and change the upstream functions, but I'm not too much for, or we can just catch both like that:
```py
except MeiliSearchApiError as err:
    return False
except requests.exceptions.RequestException as err:
    return False
```

Reply:
Sorry, I misunderstood this!
I think:
```
except MeiliSearchError:
    return False
```
is enough for the moment!

Reply:
Done